daml-lf/transaction/src/main/scala/com/daml/lf/TransactionNodeStatistics.scala

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package transaction
+
+object TransactionNodeStatistics {
+
+  /** Container for transaction statistics.
+    *
+    * @param creates number of creates nodes,
+    * @param consumingExercisesByCid number of consuming exercises by contract ID nodes,
+    * @param nonconsumingExercisesByCid number of non-consuming Exercises by contract ID nodes,
+    * @param consumingExercisesByKey number of consuming exercise by contract key nodes,
+    * @param nonconsumingExercisesByKey number of non-consuming exercise by key nodes,
+    * @param fetchesByCid number of fetch by contract ID nodes,
+    * @param fetchesByKey number of fetch by key nodes,
+    * @param lookupsByKey number of lookup by key nodes,
+    * @param rollbacks number of rollback nodes.
+    */
+  final case class Actions(
+      creates: Int,
+      consumingExercisesByCid: Int,
+      nonconsumingExercisesByCid: Int,
+      consumingExercisesByKey: Int,
+      nonconsumingExercisesByKey: Int,
+      fetchesByCid: Int,
+      fetchesByKey: Int,
+      lookupsByKey: Int,
+      rollbacks: Int,
```

Comment:
I don’t think rollbacks should be in here. First they are not actions, second it doesn’t make sense to split those into committed& non committed.

Reply:
The current metering requirements says event count should be calculated _"including the children of the rollback node but not the rollback node"_.  Also that fact there was at least one rollback will be evident by the rollback actions being non-empty.
 => removing.

Reply:
Removed

Reply:
@moritzkiefer-da - I may of misunderstood.  Is it better that a single rollback count now lives at the TransactionNodeStatistics level alongside the commit and rolledBack actions ?

Reply:
if we want to store it we should do that but given that you don’t need it and this API only exists for your usecase jsut kliling it seems fine.